### PR TITLE
Fix setting new stroke order

### PR DESCRIPTION
**Pull request by Aaron2464**
Open setStrokOrder() in order to reset data in controller.

### CODE REVIEW

File: lib/strokeOrderAnimationController.dart [hunk 1]
```diff
@@ -363,7 +363,7 @@ class StrokeOrderAnimationController extends ChangeNotifier {
   }
 
   void setStrokeOrder(String strokeOrder) {
-    final parsedJson = json.decode(_strokeOrder.replaceAll("'", '"'));
+    final parsedJson = json.decode(strokeOrder.replaceAll("'", '"'));
```
Comment (Mr-Pepe) [inline lib/strokeOrderAnimationController.dart:366]: Good catch! 
However _strokeOrder will be out of sync afterwards. It's not really used anywhere and doesn't have a getter, but I thought it's a good idea to keep the original json string around.
Can you write a getter "strokeOrder" that exposes _strokeOrder, set it here to be in sync and add it to the test?

File: test/strokeOrderAnimationController_test.dart [hunk 2]
```diff
@@ -22,6 +22,19 @@ void main() {
...
+    controllers[2].setStrokeOrder(strokeOrders[3]);
+    controllers[3].setStrokeOrder(strokeOrders[4]);
+    controllers[4].setStrokeOrder(strokeOrders[5]);
+    controllers[5].setStrokeOrder(strokeOrders[0]);
+    expect(controllers[0].nStrokes, 7);
+    expect(controllers[1].nStrokes, 10);
+    expect(controllers[2].nStrokes, 3);
+    expect(controllers[3].nStrokes, 8);
+    expect(controllers[4].nStrokes, 3);
+    expect(controllers[5].nStrokes, 5);
```
Comment (Mr-Pepe) [inline test/strokeOrderAnimationController_test.dart:37]: Good addition, but please split this into two tests. One that checks for nStrokes and strokeOrder (the new getter you should write) after initialization and one that checks for nStrokes and strokeOrder after setting new strokes.

**Comment by Mr-Pepe** [comment]
Fixes #17

**Comment by Mr-Pepe** [comment]
I can also do it myself, if you give me permission to do (see [here](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)).

File: lib/strokeOrderAnimationController.dart (hunk 1 above)
Comment (Aaron2464) [reply lib/strokeOrderAnimationController.dart:366]: `  void setStrokeOrder(String strokeOrder) {
    _strokeOrder = strokeOrder;
    final parsedJson = json.decode(_strokeOrder.replaceAll("'", '"'));
`

Do you mean like this? And have getter 'String get strokeOrder => _strokeOrder;'.

**Comment by Aaron2464** [comment]
I assume that you already can modify it whenever you want. Thanks a lot.

**Pull request merged**